src/debugger/managedcallback.cpp

```diff
@@ -761,6 +761,35 @@ HRESULT STDMETHODCALLTYPE ManagedCallback::NameChange(ICorDebugAppDomain *pAppDo
 HRESULT STDMETHODCALLTYPE ManagedCallback::UpdateModuleSymbols(ICorDebugAppDomain *pAppDomain, ICorDebugModule *pModule, IStream *pSymbolStream)
 {
     LogFuncEntry();
+
```

Comment:
1. https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/debugging/icordebugmanagedcallback-updatemodulesymbols-method
As you could see in MS docs, you should care about 2 cases here: delta pdb (for pdb data partial update for EnC/HotReload) and pdb (for pdb replace). Both have different behaviors.

2. Create temporary buffer here is wrong logic. `IStream` should be provide with callback (with data read implementation for this type) into delegate instead.

Reply:
Can you share any code to generate UpdateModuleSymbols event with delta pdb?

Reply:
I don't have it. Plus, I don't even sure about proper logic for both cases (update pdb and replace pdb), all should be analyze carefully.

Note, you current "replace pdb" logic is wrong for sure, since it don't count that previous pdb data related work should be reseted first. For example, initially was loaded wrong pdb (user just have pdb file with dll's name, from some previous built by old code) and after that load in-memory pdb... source line breakpoints that were resolved by wrong (previous pdb data) must be removed before source line breakpoints resolve in new (in-memory) pdb. Probably should be done some additional work with async-step related data and even more.